### PR TITLE
improve Keycard login performance

### LINKED ISSUE #10259: Improve keycard login time
This is a place holder to follow-up this issue from the keycard project board. 

loging in a multi account with keycard takes about 2.84 seconds (test doen with nightly from 30/03/2020) and a JCOP4 card. Eventhough that's acceptable, it could be made better by improving the interaction with the card: the code contains some unneccessary code to write to the card. 

Issue here: https://github.com/status-im/react-native-status-keycard/pull/23

Discussion here : https://discordapp.com/channels/624205794384281629/626364653890437140/694545012804812850

Video of logging time with nightly from 30.03.2020 and a JCOP4 (phone is a oneplus 6T): https://drive.google.com/open?id=1sq9dd5wJWwQCiNgVxo4xOxpIj5znvSwv

### PR DESCRIPTION
**Pull request by bitgamma**
Make login with Keycard faster. Reduces time interacting with the card from about 3,5 seconds to about 1 second (measured on OnePlus 6)

Fixes: #10259 

status: ready

### CODE REVIEW

File: src/status_im/keycard/card.cljs [hunk 1]
```diff
@@ -336,6 +336,25 @@
        (re-frame/dispatch [:keycard.callback/on-delete-error
                            (error-object->map response)]))})))
 
+(defn import-keys [{:keys [on-success] :as args}]
+  (log/info "[keycard] import-keys")
+  (keycard/import-keys
+   card
+   (merge
```
Comment (cammellos) [inline src/status_im/keycard/card.cljs]: you can use `assoc` here

File: src/status_im/keycard/card.cljs [hunk 2]
```diff
@@ -336,6 +336,25 @@
        (re-frame/dispatch [:keycard.callback/on-delete-error
                            (error-object->map response)]))})))
 
+(defn import-keys [{:keys [on-success] :as args}]
+  (log/info "[keycard] import-keys")
```
Comment (cammellos) [inline src/status_im/keycard/card.cljs]: maybe debug?

File: src/status_im/keycard/card.cljs [hunk 3]
```diff
@@ -336,6 +336,25 @@
...
+    args
+    {:on-success
+     (fn [response]
+       (log/info "[keycard response succ] import-keys")
+       (re-frame/dispatch
+        [(or on-success :keycard.callback/on-generate-and-load-key-success)
+         response]))
+     :on-failure
+     (fn [response]
+       (log/info "[keycard response fail] import-keys"
```
Comment (cammellos) [inline src/status_im/keycard/card.cljs]: `log/warn` / `log/error`

File: src/status_im/keycard/simulated_keycard.cljs [hunk 4]
```diff
@@ -307,6 +307,35 @@
...
+           {:key-uid               (get-in @state [:application-info :key-uid])
+            :encryption-public-key (ethereum/sha3 pin)}))))
+    (do
+      (log/debug "Incorrect PIN" pin)
+      (swap! state update-in
+             [:application-info :pin-retry-counter]
+             (fnil dec 3))
+      (later
+       #(on-failure
+         #js {:code    "EUNSPECIFIED"
```
Comment (cammellos) [inline src/status_im/keycard/simulated_keycard.cljs]: maybe we could define this?

**Review by cammellos — APPROVED** [review]

**Comment by cammellos** [comment]
looks good to me, just a few minor neatpicks

**Comment by bitgamma** [comment]
> looks good to me, just a few minor neatpicks

@cammellos thanks for the review! I addressed the issued you pointed out

**Comment by bitgamma** [comment]
@churik could you please test it, if you think additional tests are required? What I have tested on my Android 10 and iOS 11 devices is:

1) Generate a new account (stored on Keycard)
2) Recover account from existing Keycard
3) Login
4) Sign tx, messages, typed data

Thanks!

**Comment by Serhy** [comment]
Thanks a lot @bitgamma ! No regressions found. What has been tested is (GalaxyS20 Android11, iOS14 iPhoneX):

- Create new account
- Recover + keycard as storage
- Move multiaccount to keycard (Android only)
- Change PIN
- Recover keycard account on different device 
- Keycard is frozen scenario
- Send TX from Wallet and DApp

In respect to the performance improvement, indeed, on Galaxy J7 (slow-mid device) login time improved ~2.5 times (Develop build it's 5 sec, - while this PR build it's 2 sec)
Failed one e2e test is a known issue.

**Pull request merged**